en/development/testing.rst

```diff
@@ -1011,6 +1011,62 @@ assertions in `TestSuite
 found in `PHPUnit
 <https://phpunit.de/manual/current/en/appendixes.assertions.html>`__.
 
+Comparing test results to a file
+--------------------------------
+
+For some types of test, it may be easier to compare the result of a test to the
+contents of a file - for example, when testing the rendered output of a view.
+The ``StringCompareTrait`` adds a simple assert method for this purpose.
+
+Usage involves using the trait, setting the comparison base path and calling
+``assertSameAsFile``::
+
+    use Cake\TestSuite\StringCompareTrait;
+    use Cake\TestSuite\TestCase;
+
+    class SomeTest extends TestCase
+    {
+        use StringCompareTrait;
+
+        public function setUp()
+        {
+            $this->_compareBasePath = APP . 'tests/comparisons/';
+            parent::setUp();
+        }
+
+        public function testExample()
+        {
+            $result = ...;
+            $this->assertSameAsFile('example.php', $result);
```

Comment:
Not 100% sure what you mean - `assertSameAsFile` expects only a filename.

I'll update the example to use the DS constant for compareBasePath, as [existing usage of this assert method does](https://github.com/cakephp/bake/blob/20c48a87150b50063ab0e8bec46436edd401c45e/tests/TestCase/View/BakeViewTest.php#L42).